src/plugins/jartic/datasource.py

```
@@ -0,0 +1,268 @@
+    ) -> List[Dict[str, Any]]:
+        archive_index = await self.downloader.get_archive_index()
+
+        archives_in_range = []
+        for archive in archive_index:
+            archive_year = archive['year']
+            archive_month = archive['month']
+
+            if (archive_year > start_date.year or
+                (archive_year == start_date.year and archive_month >= start_date.month)):
+                if (archive_year < end_date.year or
+                    (archive_year == end_date.year and archive_month <= end_date.month)):
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)